### PR TITLE
Adding isError property in Checkbox

### PR DESCRIPTION
**Pull request by Ahmet-K**
Adding isError property into `Checkbox` component.

Tasks:
- [X] test
- [X] storybook documentation
- [X] storybook unstyled
- [X] storybook classBased

https://github.com/Capgemini/dcx-react-library/issues/581

### CODE REVIEW

File: src/common/components/commonTypes.ts [hunk 1]
```diff
@@ -274,6 +274,19 @@ export type FormRadioCheckboxProps = {
...
+  isError?: boolean;
+  // /**
+  //  * error details including text, class name, ID, and visually hidden text.
+  //  */
+  // error?: {
+  //   text: string;
+  //   className: string;
+  //   id: string;
+  //   visuallyHiddenText: string;
+  // };
```
Comment (daniele-zurico) [inline src/common/components/commonTypes.ts:289]: ?

File: src/formCheckbox/FormCheckbox.tsx [hunk 2]
```diff
@@ -46,8 +58,9 @@ export const FormCheckbox = ({
...
     hint={hint}
     nested={nested}
-    inputClassName={inputClassName}
-    labelClassName={labelClassName}
-    itemClassName={itemClassName}
-  />
-);
+    itemClassName={`${itemClassName} ${errorClasses.container}`}
+    inputClassName={`${inputClassName} ${errorClasses.checkbox}`}
+    labelClassName={`${labelClassName} ${errorClasses.label}`}
```
Comment (daniele-zurico) [inline src/formCheckbox/FormCheckbox.tsx:63]: use the `classNames` utils please

File: src/formCheckbox/__test__/FormCheckbox.test.tsx [hunk 3]
```diff
@@ -6,7 +6,7 @@ import userEvent from '@testing-library/user-event';
 import { act } from 'react-dom/test-utils';
 import * as hooks from '../../common/utils/clientOnly';
 
-describe('FormCheckbox', () => {
+describe.only('FormCheckbox', () => {
```
Comment (daniele-zurico) [inline src/formCheckbox/__test__/FormCheckbox.test.tsx:9]: remove .only

File: src/formCheckbox/__test__/FormCheckbox.test.tsx [hunk 4]
```diff
@@ -495,4 +495,72 @@ describe('FormCheckbox', () => {
...
+        id="myId"
+        name="group1"
+        value="choice 1"
+        label="my label"
+        onChange={handleChange}
+        isError={true}
+      />
+    );
+
+    console.log("OUTPUT", container.innerHTML);
```
Comment (daniele-zurico) [inline src/formCheckbox/__test__/FormCheckbox.test.tsx:513]: remove it

File: src/common/components/commonTypes.ts (hunk 1 above)
Comment (Ahmet-K) [reply src/common/components/commonTypes.ts]: Done.

File: src/formCheckbox/FormCheckbox.tsx (hunk 2 above)
Comment (Ahmet-K) [reply src/formCheckbox/FormCheckbox.tsx]: Done.

File: src/formCheckbox/__test__/FormCheckbox.test.tsx (hunk 3 above)
Comment (Ahmet-K) [reply src/formCheckbox/__test__/FormCheckbox.test.tsx]: Done.

File: src/formCheckbox/__test__/FormCheckbox.test.tsx (hunk 4 above)
Comment (Ahmet-K) [reply src/formCheckbox/__test__/FormCheckbox.test.tsx]: Done

**Comment by daniele-zurico** [comment]
@Ahmet-K in the storybook documentation please add one example for `classBased`.
You can make the checkbox and the label red for example (something like when you need to accept terms & condition), and in the description you can say that is not a standard gds standard behaviour but the library allow to do so

**Review by daniele-zurico — APPROVED** [review]

**Pull request merged**